docs/source/best_practices.rst

```diff
@@ -0,0 +1,244 @@
+
+    CREATE OR REPLACE TABLE seattle_stations WITH (
+        location = '/data/seattle_stations/*.parquet',
+        gpu=True
+    )
+
+
+.. code-block:: sql
+
+    CREATE OR REPLACE TABLE city_stations AS
+    SELECT * FROM atlanta_stations
+    UNION ALL
+    SELECT * FROM seattle_stations
+
+you get a new table that has two partitions. Then if you use it in a join
+
+.. code-block:: sql
+
+    SELECT
+        yr,
+        city,
+        CASE WHEN city='Atlanta' THEN
+            sum(inches)/{atl_stations}
+        ELSE
+            sum(inches)/{seat_stations}
+        END AS inches
+    FROM precip
+    JOIN city_stations
+    ON precip.station_id = city_stations.station_id
+    GROUP BY yr, city
+    ORDER BY yr ASC
+
+Dask-SQL won't perform a broadcast join and will instead perform a traditional join with a corresponding slow compute time.
+However, if you were to repartition the smaller table to a single partition and rerun the operation
+
+.. code-block:: python
+
+    c.create_table("city_stations", c.sql("select * from city_stations").repartition(npartitions=1))
+
+.. code-block:: sql
+
+    SELECT
+        yr,
+        city,
+        CASE WHEN city='Atlanta' THEN
+            sum(inches)/{atl_stations}
+        ELSE
+            sum(inches)/{seat_stations}
+        END AS inches
+    FROM precip
+    JOIN city_stations
+    ON precip.station_id = city_stations.station_id
+    GROUP BY yr, city
+    ORDER BY yr ASC
+
+Dask-SQL is able to recognize this as a broadcast join and the result is a significantly faster compute time.
+
+Optimize Partition Sizes for GPUs
+---------------------------------
+File formats like `Apache ORC <https://orc.apache.org/>`_ and `Apache Parquet <https://parquet.apache.org/>`_ are designed so that they can be pulled from disk and be deserialized by CPUs quickly.
+However, loading data into GPUs has a substantial additional cost in the form of transfers from CPU to GPU memory.
+Minimizing that cost is often achieved by increasing partition size.
+Even when using Dask-SQL on GPUs, upstream CPU systems will likely produce small files resulting in small partitions.
+It's worth taking the time to repartition to larger partition sizes before querying the files on GPUs, especially when querying the same files multiple times.
+
+There's no single optimal size so choose a size that's tuned for your workflow.
+Operations like joins and concatenations greatly increase GPU memory utilization, even if temporarily, but if you're not performing many of these operations, the larger the partition size the better.
+Larger partition sizes increase disk to GPU throughput and keep GPU utilization higher for faster runtimes.
+
+We recommend a starting point of around 2gb uncompressed data per partition for GPUs.
+It's usually not necessary to change from default settings when running Dask-SQL on CPUs, but if you want to manually set partition sizes, we've found 128-256mb per partition to be a good starting place.
```

Comment:
Once we have a page linking to a couple of example notebooks (weather and taxi) we should have a link to that page from here as way to explore examples